hedy.py

```diff
@@ -2830,6 +2814,168 @@ def print_empty_brackets(self, meta, args):
         return self.print(meta, args)
 
 
+@v_args(meta=True)
+@hedy_transpiler(level=1, microbit=True)
+@source_map_transformer(source_map)
+class MicrobitConvertToPython_1(ConvertToPython_1):
+    def print(self, meta, args):
+        # escape needed characters
+        argument = process_characters_needing_escape(args[0])
+        return f"display.scroll('{argument}')"
+
+
+@v_args(meta=True)
+@hedy_transpiler(level=2, microbit=True)
+@source_map_transformer(source_map)
+class MicrobitConvertToPython_2(MicrobitConvertToPython_1, ConvertToPython_2):
+    def print(self, meta, args):
+        args_new = [self.make_print_ask_arg(a, meta) for a in args]
+        argument_string = ' '.join(args_new)
+        return f"display.scroll({argument_string})"
+
+    def make_print_ask_arg(self, arg, meta, var_to_escape=''):
+        if self.is_variable(arg, meta.line):
+            return arg
+        else:
+            # If the argument is not a variable, return it as a string literal with quotes
+            return f"'{arg}'"
+
+    def sleep(self, meta, args):
+        if not args:
+            return f"sleep(1000){self.add_debug_breakpoint()}"  # Default 1 second sleep in milliseconds
+        else:
+            value = args[0]
+            if self.is_int(value):
+                # Direct conversion of seconds to milliseconds
+                milliseconds = int(value) * 1000
+            else:
+                # If the value is a variable, ensure it's used correctly
+                milliseconds = f"{value} * 1000"
+            return f"    sleep({milliseconds}){self.add_debug_breakpoint()}"
```

Comment:
I think you are adding whitespace in front of commands because the whole program will need to be placed in a `while True:` loop. Is this correct? If so, I am afraid adding whitespace here will not work. Instead, you need to transpile the whole program and then indent it right before you append it to the existing `while True:` program.

Reply:
As discussed in out meeting I'll get rid of the white spaces in the transpiler and I'll handle them in Hedy.app instead.